### PR TITLE
Weapon

### PR DESCRIPTION
**Pull request by fsandel**
- implemented a weapon and torch sprite
- weapon can shoot enemies and has kick
- torch lights up the room
- implemented destructible walls

### CODE REVIEW

File: src/free_utils.c [hunk 1]
```diff
@@ -55,11 +73,13 @@ void	free_window_struct(t_window *window)
 	mlx_delete_texture(window->map->health_text[1]);
 	mlx_delete_texture(window->map->exit_text[0]);
 	mlx_delete_texture(window->map->exit_text[1]);
+	mlx_delete_texture(window->map->destructible_tex);
```
Comment (philippschwarzmueller) [inline src/free_utils.c:76]: is this failsafe if the map hasn't got any destructible walls?

File: src/setup_player.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include <cub3D.h>
+
+static t_weapon	*setup_weapon(void);
+
+t_player	*setup_player(t_map *map)
+{
+	t_player	*player;
+
+	player = malloc(sizeof(t_player));
```
Comment (philippschwarzmueller) [inline src/setup_player.c:9]: what about unprotected mallocs, do we want to accept these?

File: src/setup_player.c [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	player->dir = map->start_dir;
+	player->weapon = setup_weapon();
+	return (player);
+}
+
+static t_weapon	*setup_weapon(void)
+{
+	t_weapon	*weapon;
+
+	weapon = malloc(sizeof(t_weapon));
```
Comment (philippschwarzmueller) [inline src/setup_player.c:22]: what about unprotected mallocs, do we want to accept these?

**Review by philippschwarzmueller — APPROVED** [review]
lgtm tried it out locally and really love the torch feature

File: src/setup_player.c (hunk 2 above)
Comment (fsandel) [reply src/setup_player.c:9]: My Idea was to go through the Code at the end of production and add a malloc check + free after every malloc, so we don't have to constantly refactor all the dynamic frees

**Pull request merged**